stdlib/builtins.pyi

```diff
@@ -334,13 +334,13 @@ class str(Sequence[str]):
     def __new__(cls: Type[_T], o: bytes, encoding: str = ..., errors: str = ...) -> _T: ...
     def capitalize(self) -> str: ...
     def casefold(self) -> str: ...
-    def center(self, __width: int, __fillchar: str = ...) -> str: ...
+    def center(self, __width: SupportsIndex, __fillchar: str = ...) -> str: ...
     def count(self, x: str, __start: Optional[SupportsIndex] = ..., __end: Optional[SupportsIndex] = ...) -> int: ...
     def encode(self, encoding: str = ..., errors: str = ...) -> bytes: ...
     def endswith(
         self, __suffix: Union[str, Tuple[str, ...]], __start: Optional[SupportsIndex] = ..., __end: Optional[SupportsIndex] = ...
     ) -> bool: ...
-    def expandtabs(self, tabsize: int = ...) -> str: ...
+    def expandtabs(self, tabsize: SupportsIndex = ...) -> str: ...
```

Comment:
```
Python 3.7.2 (default, Jan  8 2019, 13:38:59) 
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo:
...     def __index__(self):
...             return 2
... 
>>> "\t\tx".expandtabs(Foo())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: an integer is required (got type Foo)
```

Reply:
I did a bit of digging and this appears to be a new feature in python 3.8. From the [release notes](https://docs.python.org/3/whatsnew/3.8.html#build-and-c-api-changes):
> Functions that convert Python number to C integer like `PyLong_AsLong()` and argument parsing functions like `PyArg_ParseTuple()` with integer converting format units like `'i'` will now use the `__index__()` special method instead of `__int__()`, if available. The deprecation warning will be emitted for objects with the `__int__()` method but without the `__index__()` method (like `Decimal` and `Fraction`). `PyNumber_Check()` will now return 1 for objects implementing `__index__()`. `PyNumber_Long()`, `PyNumber_Float()` and `PyFloat_AsDouble()` also now use the `__index__()` method if available. (Contributed by Serhiy Storchaka in bpo-36048 and bpo-20092.)

What would be the best course of action here?
Ignore the version bound and use `SupportsIndex` anyway?
Add a variable that defaults that defaults to `int` for <3.8 and `SupportsIndex` for later versions?

Reply:
You could use a type alias:

```python3
if sys.version_info >= (3, 8):
    _IndexIn38 = SupportsIndex
else:
    _IndexIn38 = int
```

Reply:
After some further testing it seems that only the `expandtabs` is the odd exception here,
all the other have support for `SupportsIndex` prior to 3.8.